cpp/src/arrow/array/array_primitive.h

```diff
@@ -141,6 +148,14 @@ class ARROW_EXPORT DayTimeIntervalArray : public PrimitiveArray {
   // For compatibility with Take kernel.
   TypeClass::DayMilliseconds GetView(int64_t i) const { return GetValue(i); }
 
+  IteratorType begin() const { return IteratorType(*this); }
+
+  IteratorType end() const { return IteratorType(*this, length()); }
```

Comment:
Thanks for adding this. Is it being exercised in the tests?

Reply:
yes, the new tests use this.